main/src/library/Channel/Channel.flix

```diff
@@ -0,0 +1,582 @@
+            let channelIndexPairs = MutList.new();
+            Array.foreachWithIndex(
+                (c, i) -> MutList.push!(newChannelIndexPair(c, i), channelIndexPairs),
+                channels
+                );
+
+
+            // TODO : Randomize the order channels are looked at.
+            // This prevents backpressure from building up on one channel.
+            /// Collections.shuffle(channelIndexPairs, ThreadLocalRandom.current());
+
+            // Find channels with waiting elements
+            MutList.foreachWithIndex(
+                (cip, _) -> {
+                    let e_opt = tryGet(getChannel(cip));
+
+                    match e_opt {
+                        case None => ()
+                        case Some(element) => {
+                            // There is a waiting element in this channel.
+                            // Return the element and the branchNumber of this channel
+
+                            // If an element found, add a new selectChoice to the MutList in order to later return
+                            let selectChoice = Some(newSelectChoice(getIndex(cip), element, false));
+                            MutList.push!(selectChoice, selectChoices)
+                        }
+                    }
+                },
+                channelIndexPairs
+                );
+
+
+            // Element found
+            if (!MutList.isEmpty(selectChoices)) {
+                Console.printLine("Selected an element");
+
+                MutList.get(0, selectChoices)}
+            // No element was found but if there is a default case, choose this
+            else if (hasDefault) {
+                Console.printLine("Default selection");
+                None
+            }
+            // No element was found.
+            else {
+                Console.printLine("Nothing selected");
+                // Add our condition to all channels to get notified when a new element is added
+                Array.foreach(
+                    (c) -> {
+                        addGetter(c, selectLock, condition)
+                    },
+                    channels);
+
+                // Unlock all channels in sorted order, so other threads may input elements
+                unlockAllChannels(sortedChannels);
+
+                // Wait for an element to be added to any of the channels
+                java_await(condition);
+
+                // Unlock the selectLock, which is relevant when a different thread wants to put
+                // an element into a channel that was not selected from the select.
+                // This other channel will then signal the condition from selectLock (in the put method),
+                // so it needs the lock.
+                java_unlock(selectLock);
+
+                select_helper(channels, sortedChannels, selectLock, condition, hasDefault)
+            }
+        }
+
+    // TODO : -implement spawn
```

Comment:
I think spawn has to "stay" in the compiler. The semantics of "spawn" cannot be expressed in Flix itself.

Less work for you ;)